stdlib/templates/hashtbl.template.mli

```diff
@@ -0,0 +1,506 @@
+   by [f] during the iteration.
+*)
+
+val filter_map_inplace: f:(key:'a -> data:'b -> 'b option) -> ('a, 'b) t ->
+    unit
+(** [Hashtbl.filter_map_inplace ~f tbl] applies [f] to all bindings in
+    table [tbl] and update each binding depending on the result of
+    [f].  If [f] returns [None], the binding is discarded.  If it
+    returns [Some new_val], the binding is update to associate the key
+    to [new_val].
+
+    Other comments for {!iter} apply as well.
+    @since 4.03.0 *)
+
+val fold : f:(key:'a -> data:'b -> 'c -> 'c) -> ('a, 'b) t -> init:'c -> 'c
+(** [Hashtbl.fold ~f tbl ~init] computes
+   [(f kN dN ... (f k1 d1 init)...)],
+   where [k1 ... kN] are the keys of all bindings in [tbl],
+   and [d1 ... dN] are the associated values.
+   Each binding is presented exactly once to [f].
+
+   The order in which the bindings are passed to [f] is unspecified.
+   However, if the table contains several bindings for the same key,
+   they are passed to [f] in reverse order of introduction, that is,
+   the most recent binding is passed first.
+
+   If the hash table was created in non-randomized mode, the order
+   in which the bindings are enumerated is reproducible between
+   successive runs of the program, and even between minor versions
+   of OCaml.  For randomized hash tables, the order of enumeration
+   is entirely random.
+
+   The behavior is not defined if the hash table is modified
+   by [f] during the iteration.
+*)
+
+val length : ('a, 'b) t -> int
+(** [Hashtbl.length tbl] returns the number of bindings in [tbl].
+   It takes constant time.  Multiple bindings are counted once each, so
+   [Hashtbl.length] gives the number of times [Hashtbl.iter] calls its
+   first argument. *)
+
+val randomize : unit -> unit
+(** After a call to [Hashtbl.randomize()], hash tables are created in
+    randomized mode by default: {!create} returns randomized
+    hash tables, unless the [~random:false] optional parameter is given.
+    The same effect can be achieved by setting the [R] parameter in
+    the [OCAMLRUNPARAM] environment variable.
+
+    It is recommended that applications or Web frameworks that need to
+    protect themselves against the denial-of-service attack described
+    in {!create} call [Hashtbl.randomize()] at initialization
+    time.
+
+    Note that once [Hashtbl.randomize()] was called, there is no way
+    to revert to the non-randomized default behavior of {!create}.
+    This is intentional.  Non-randomized hash tables can still be
+    created using [Hashtbl.create ~random:false].
+
+    @since 4.00.0 *)
+
+val is_randomized : unit -> bool
+(** Return [true] if the tables are currently created in randomized mode
+    by default, [false] otherwise.
+    @since 4.03.0 *)
+
+val rebuild : ?random (* thwart tools/unlabel *) :bool ->
+    ('a, 'b) t -> ('a, 'b) t
+(** Return a copy of the given hashtable.  Unlike {!copy},
+    [{!rebuild} h] re-hashes all the (key, value) entries of
```

Comment:
```suggestion
    {!rebuild}[ h] re-hashes all the (key, value) entries of
```
I think? The `{!rebuild}` appears in the output (it was wrong in the original PR)

Reply:
Fixed. (By the way, does Github email you every time I write 'Fixed'? If so, I'll stop doing it!)

Reply:
It does, but that's my choice to have the notification on (and I don't mind!). For my preference, I prefer the comment to "Resolve conversation" (which doesn't generate a notification). GitHub also sets lots of headers describing the notification, so it doesn't clog my "inbox" per se...